src/copilot/inspect/InspectionCache.ts

```diff
@@ -0,0 +1,94 @@
+import { SymbolKind, TextDocument } from 'vscode';
+import { METHOD_KINDS, getClassesAndMethodsOfDocument, logger } from '../utils';
+import { Inspection } from './Inspection';
+import * as crypto from "crypto";
+import { SymbolNode } from './SymbolNode';
+
+/**
+ * A map based cache for inspections of a document.
+ * format: `Map<documentKey, Map<symbolQualifiedName, [symbolVersionId, Promise<Inspection[]>]`
+ */
+const DOC_SYMBOL_VERSION_INSPECTIONS: Map<string, Map<string, [string, Promise<Inspection[]>]>> = new Map();
```

Comment:
You are right.
in my old implementation, the `inspecting` process is saved, so that next call will get the promise directly when first request is still processing (instead of sending another request). but it's no need now